src/compiler/checker.ts

```diff
@@ -11484,6 +11484,7 @@ namespace ts {
                 case SyntaxKind.FunctionExpression:
                 case SyntaxKind.ArrowFunction:
                 case SyntaxKind.MethodDeclaration:
+                case SyntaxKind.FunctionDeclaration: // Function declarations can have context when annotated with a jsdoc @type

```

Comment:
I couldn't find a test with an example, and couldn't come up with one myself. I didn't think `@type` was treated as a contextual type.

Reply:
There's a trio of jsdoc tests that demonstrate it which were borked with the better handling of contextual mappers and context checking (ie, they abused how the different entrypoint cached a different type to force a type to be cached from the jsdoc annotation) without proper handling here. The responsible codepath is definitely doing an unsafe cast, technically, since this function isn't even declared to accept a `FunctionDeclaration`, however checking it is perfectly reasonable when an annotated declaration is viewed in the same light as an annotated assigned expression.

Reply:
can you point me to the tests?